packages/backend/src/open_payments/grant/service.test.ts

```diff
@@ -132,5 +137,71 @@ describe('Grant Service', (): void => {
         })
       ).resolves.toBeUndefined()
     })
+
+    test('cannot store grant with misformatted management url', async (): Promise<void> => {
+      const options: CreateOptions = {
+        accessToken: uuid(),
+        managementUrl: '',
+        authServer: faker.internet.url(),
+        accessType: AccessType.IncomingPayment,
+        accessActions: [AccessAction.ReadAll]
+      }
+      await expect(grantService.create(options)).rejects.toThrow(
+        'invalid management id'
+      )
+    })
+  })
+
+  describe('Update Grant', (): void => {
+    let options: CreateOptions
+    let grant: Grant
+    let authServerId: string
+    beforeEach(async (): Promise<void> => {
+      options = {
+        authServer: faker.internet.url(),
+        accessType: AccessType.IncomingPayment,
+        accessActions: [AccessAction.ReadAll],
+        accessToken: uuid(),
+        managementUrl: `${faker.internet.url()}/gt5hy6ju7ki8`,
+        expiresIn: 3000
+      }
+      grant = await grantService.create(options)
+      const authServerService = await deps.use('authServerService')
+      authServerId = (await authServerService.getOrCreate(options.authServer))
+        .id
+    })
+
+    test('can update grant', async (): Promise<void> => {
+      const updateOptions = {
+        accessToken: uuid(),
+        managementUrl: `${faker.internet.url()}/${uuid()}`,
+        expiresIn: 6000
+      }
+      const updatedGrant = await grantService.update(grant, updateOptions)
+      expect(updatedGrant).toMatchObject({
```

Comment:
This could be a `toEqual` compared against `grant` (instead of `options`)

Reply:
But then we'll also have to check the entire authServer details plus createdAt, updatedAt. I think it's enough to check the crucial details.

Reply:
```
      expect(updatedGrant).toEqual({
        ...grant,
        accessToken: updateOptions.accessToken,
        managementId: updateOptions.managementUrl.split('/').pop(),
        expiresAt: new Date(Date.now() + updateOptions.expiresIn * 1000),
        updatedAt: updatedGrant.updatedAt
      })
```

Reply:
Ok, you win :joy: